lib/class/api.class.php

```diff
@@ -172,7 +175,12 @@ public static function handshake($input)
                     ($timestamp > (time() + 1800))) {
                     debug_event('API', 'Login Failed: timestamp out of range ' . $timestamp . '/' . time(), 1);
                     AmpError::add('api', T_('Login Failed: timestamp out of range'));
-                    echo XML_Data::error('401', T_('Error Invalid Handshake - ') . T_('Login Failed: timestamp out of range'));
+                    if ($outputFormat == 'json') {
+                      echo JSON_Data::error('401', T_('Error Invalid Handshake - ') . T_('Login Failed: timestamp out of range'));
+                    }
+                    elseif ($outputFormat == 'xml') {
```

Comment:
I would personally rewrite it as:

```
if ($outputFormat == 'json') {
…
} else {  // Any other format, defaults to XML
…
}
```

So that if one adds a new output format, the XML default will be made explicit.